ziti/cmd/ascode/upload/upload.go

```diff
@@ -0,0 +1,466 @@
+
+	edge.AddLoginFlags(cmd, &u.loginOpts)
+	u.loginOpts.Out = out
+	u.loginOpts.Err = errOut
+
+	return cmd
+}
+
+func (u *Upload) Init() {
+	u.verbose = u.loginOpts.Verbose
+
+	logLvl := logrus.InfoLevel
+	if u.verbose {
+		logLvl = logrus.DebugLevel
+	}
+
+	pfxlog.GlobalInit(logLvl, pfxlog.DefaultOptions().Color())
+	internal.ConfigureLogFormat(logLvl)
+
+	client, err := mgmt.NewClient()
+	if err != nil {
+		loginErr := u.loginOpts.Run()
+		if loginErr != nil {
+			log.Fatal(err)
+		}
+		client, err = mgmt.NewClient()
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+	u.client = client
+	u.reader = FileReader{}
+}
+
+func (u *Upload) Execute(data map[string][]interface{}, inputArgs []string) (map[string]any, error) {
+
+	args := arrayutils.Map(inputArgs, strings.ToLower)
+
+	u.configCache = cache.New(time.Duration(-1), time.Duration(-1))
+	u.serviceCache = cache.New(time.Duration(-1), time.Duration(-1))
+	u.edgeRouterCache = cache.New(time.Duration(-1), time.Duration(-1))
+	u.authPolicyCache = cache.New(time.Duration(-1), time.Duration(-1))
+	u.extJwtSignersCache = cache.New(time.Duration(-1), time.Duration(-1))
+	u.identityCache = cache.New(time.Duration(-1), time.Duration(-1))
+
+	result := map[string]any{}
+	all := slices.Contains(args, "all") || len(args) == 0
+
+	cas := map[string]string{}
+	if all ||
+		slices.Contains(args, "ca") || slices.Contains(args, "cas") ||
+		slices.Contains(args, "certificate-authority") || slices.Contains(args, "certificate-authorities") {
+		if u.verbose {
+			log.
+				Debug("Processing CertificateAuthorities")
+		}
+		var err error
+		cas, err = u.ProcessCertificateAuthorities(data)
+		if err != nil {
+			return nil, err
+		}
+		if u.verbose {
+			log.
+				WithField("certificateAuthorities", cas).
+				Debug("CertificateAuthorities created")
+		}
+	}
+	result["certificateAuthorities"] = cas
+	_, _ = fmt.Fprintf(u.Err, "\u001B[2KCreated %d CertificateAuthorities\r\n", len(cas))
+
+	externalJwtSigners := map[string]string{}
+	if all ||
+		slices.Contains(args, "external-jwt-signer") || slices.Contains(args, "external-jwt-signers") ||
+		slices.Contains(args, "auth-policy") || slices.Contains(args, "auth-policies") ||
+		slices.Contains(args, "identity") || slices.Contains(args, "identities") {
+		if u.verbose {
+			log.
+				Debug("Processing ExtJWTSigners")
+		}
+		var err error
+		externalJwtSigners, err = u.ProcessExternalJwtSigners(data)
+		if err != nil {
+			return nil, err
+		}
+		if u.verbose {
+			log.
+				WithField("externalJwtSigners", externalJwtSigners).
+				Debug("ExtJWTSigners created")
+		}
+	}
+	_, _ = fmt.Fprintf(u.Err, "\u001B[2KCreated %d ExtJWTSigners\r\n", len(externalJwtSigners))
+	result["externalJwtSigners"] = externalJwtSigners
+
+	authPolicies := map[string]string{}
+	if all ||
+		slices.Contains(args, "auth-policy") || slices.Contains(args, "auth-policies") ||
+		slices.Contains(args, "identity") || slices.Contains(args, "identities") {
```

Comment:
i feel like these or's should be changed into functions and tests added for them...

* IdentitiesNeeded(args)
* AuthPoliciesNeeded(args)
* ServicesNeeded(args)

etc. i think it'll clean up the or's and it'll make it testable